homeassistant/components/homekit/accessories.py

```diff
@@ -181,7 +181,6 @@ def __init__(self, hass, driver, name):
 
     def setup_message(self):
         """Prevent print of pyhap setup message to terminal."""
-        pass
```

Comment:
IMO we should leave `pass` as it is. It might not be necessary, but it certainly improves readability.

Reply:
I agree. Seeing bare lines after a `:` gives me the willies.

Reply:
I agree

Reply:
FWIW I do not agree. But anyway I don't think there's a clean way to omit just these particular unnecessary passes, it would have to be done to all of them. And I think doing that would probably make things worse also in eyes of some people who would like to keep just the unnecessary passes that would have been removed in this PR. I certainly think that would make things worse overall. So I'm afraid we're a bit stuck with the upgrade here, then.

Reply:
You can still look at upgrading – just disable `unnecessary-pass` in `pylintrc`.

Reply:
I know, this ("would have to be done to all of them") is what I noted in my previous message, and noted also that I don't think it would be a good way forward.

Reply:
(Just in case it's not clear, there are more than these particular unnecessary passes caught by pylint, disabling all of them would disable also the ones that have been thus far accepted as good.)

Reply:
I just went through the pass commit. Of the 100 changes, **none** have been ones I would change (not just function -> pass). I believe it's save to exclude this one ;)

Reply:
(I hope we're talking about the same commit --  https://github.com/home-assistant/home-assistant/pull/18750/commits/67663f86c2951ad45b3e411f8ae1a724b41419b7 has 85 unnecessary passes removed.)

If you want to evaluate the effect of disabling unnecessary-pass globally, it's 1) **not enough** to go through the commit https://github.com/home-assistant/home-assistant/pull/18750/commits/67663f86c2951ad45b3e411f8ae1a724b41419b7 in this PR, and 2) it's not possible to add a disable that would just affect the passes in this commit. In fact, I dont think properly evaluating the effect is sanely doable at all. There are other cases around the codebase where unnecessary-pass has triggered before, and they have been addressed by developers before commit or as part of other commits (there are no clearly labeled commits that could be examined that would give a bigger picture, just a few isolated bits here and there). So disabling it globally would start to let also possibly/probably unwanted unnecessary-passes (i.e. _not_ the kind of cases addressed in this PR but other ones) creeping in again.

So I'm afraid there's nothing that could be "safely excluded" here. Unless you mean the upgrade too.

In my opinion global disabling of unnecessary-pass would be a net negative, and because its effect cannot be properly estimated, that's not something I will be personally doing/submitting at this point. If this PR is not acceptable with the unnecessary-pass commit included, I will take a look at submitting it also without the upgrades, i.e. just the remaining uncontroversial changes. But without the upgrade, that would mean there would be nothing guarding similar things being reintroduced later (although I would probably do the upgrade and future ones locally anyway and continue trying to submit needed and useful changes, but I can imagine that would become tiresome to get through reviews).

Reply:
To be honest I don't fully understand why you think excluding `pass` globally (with disable `unnecessary-pass`) is a bad idea. `pylint` is used to help developers improve their code and catch common python mistakes. However it's not perfect. Sometimes the default settings aren't the best regarding readability.

For the `pass` statement itself:
As I'm sure you know, it doesn't do much harm that would be worth preventing IMO. All errors that might be prevented since a method only contained `pass` should also be pretty easy to spot during real world testing and review. (After all, there aren't any real issues in https://github.com/home-assistant/home-assistant/pull/18750/commits/67663f86c2951ad45b3e411f8ae1a724b41419b7.)

As for the PR, I like all other changes so I think removing the aforementioned commit and adding disable `unnecessary-pass` should be enough to get it merged. We shouldn't close this, just because of `pass`.

Reply:
Pylint has a lot of opinionate cleanup-like options, it does much more than helps catch common mistakes. Many (most?) of those are enabled for HA, even though what they do have only cosmetic/aesthetic value. Opinions vary whether those are liked or are a good thing, and we've now hit one here where they vary.

As said, I don't personally think that a global `unnecessary-pass` disable would be a good thing to do, so it wouldn't make sense for me to do that. Others seem to have different opinions on the matter, so I suggest someone first does that along with their reasoning in pylintrc in a separate PR, and afterwards I'll rebase this PR, keeping only the uncontroversial changes. Will take a look at upgrading to 2.2.2 meanwhile.

Reply:
#18985

Reply:
I wish PyLint was a bit more configurable. A config option for `unnecessary-pass` to not count doc strings as statements when determining if a pass is needed would have helped here.

Reply:
Yeah, but then again I don't think there's much to blame them for here -- these `pass`es **are** unnecessary after all, people are just accustomed to seeing them. I found their flagging and looks after removal somewhat odd at first sight too, but after thinking about it for a bit, this is consistent and justifiable so I've tweaked my personal preference instead.